### PR TITLE
Improve rules and policies for testing CIRCSTORE-131

### PR DESCRIPTION
**Pull request by marcjohnson-kint**
Recently it has been mentioned that the default loan and request policies could be more useful for testing.

In order to make testing easier in the hosted environments, change the circulation rules and policies to
* allow any request types
* use rolling policy for most circumstances
* Add a recall management policy to the rolling policy

The change also removes the second example notice policy because it is the same as the first

### CODE REVIEW

File: reference-data/loan-policy-storage/loan-policies/rolling.json [hunk 1]
```diff
@@ -6,8 +6,8 @@
   "loansPolicy" : {
     "profileId" : "Rolling",
     "period" : {
-      "duration" : 1,
-      "intervalId" : "Months"
+      "duration" : 60,
+      "intervalId" : "Days"
```
Comment (kostyantyn-kh) [inline reference-data/loan-policy-storage/loan-policies/rolling.json]: What is the purpose for this change?

Comment (marcjohnson-kint) [reply reference-data/loan-policy-storage/loan-policies/rolling.json]: Cate asked for it to be made when we talked about what policy would best help them test
